### PR TITLE
docs(setup): Add pacman / Arch Linux

### PR DESCRIPTION
**Pull request by hashworks**
With the next version (since 0.24.1 isn't building for me, but master is) Vector will be available in the Arch Linux community repository.

Current [PKGBUILD](https://man.archlinux.org/man/PKGBUILD.5) file in development: https://fb.hash.works/tIUPCi/

I wasn't sure about the weights, I figured you wanted stuff to be sorted alphabetically?

### CODE REVIEW

File: website/content/en/docs/setup/installation/package-managers/pacman.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+title: Install Vector using pacman
+short: pacman
+weight: 7
+---
+
```
Comment (spencergilbert) [inline website/content/en/docs/setup/installation/package-managers/pacman.md:6]:
```suggestion

{{< info title="Community Maintained" >}}
The Vector pacman package is supported and maintained by the open source community.
{{< /info >}}

Stolen from [nixpkg](https://github.com/vectordotdev/vector/blob/2d908b566bd326f68c4ea5808d0fd8065c9fcce4/website/content/en/docs/setup/installation/package-managers/nix.md), figured it'd be good to be consistent. 


```

**Review by spencergilbert — COMMENTED** [review]
> With the next version (since 0.24.1 isn't building for me, but master is) Vector will be available in the Arch Linux community repository.

Interesting that it's not building for that version 🤔 is that something worth debugging?

> I wasn't sure about the weights, I figured you wanted stuff to be sorted alphabetically?

Alphabetical makes sense to me 👍 

Appreciate pulling this together!

File: website/content/en/docs/setup/installation/package-managers/pacman.md (hunk 1 above)
Comment (hashworks) [reply website/content/en/docs/setup/installation/package-managers/pacman.md:6]: Sounds good, added in 38a051212.

**Comment by hashworks** [comment]
> Interesting that it's not building for that version  is that something worth debugging?

No. The error is `associated function write_record is never used`, referencing [this function](https://github.com/vectordotdev/vector/blob/v0.24.1/lib/vector-buffers/src/variants/disk_v2/writer.rs#L585). This happens because it's only used in tests and the project doesn't allow dead code - so basically it is a nitpick by cargo. In the [current master](https://github.com/vectordotdev/vector/blob/master/lib/vector-buffers/src/variants/disk_v2/writer.rs#L585) this is fixed by adding `#[cfg(test)]` (see [this discussion](https://github.com/vectordotdev/vector/pull/14520#discussion_r977854947)).

**Comment by hashworks** [comment]
I've fixed the newline-check for `pacman.md`.

**Review by spencergilbert — APPROVED** [review]
@jszwedko want me to mark this to get picked into 0.25, or do we want to be double sure the package builds properly?

**Review by bruceg — APPROVED** [review]

**Comment by hashworks** [comment]
0.25.0 released to community. https://archlinux.org/packages/community/x86_64/vector/

**Pull request merged**